config-UI: Standard-Fahrzeug nicht änderbar

### Describe the bug

Ich hatte bei der Erstellung des Loadpoint `db:5` die "automatischen Erkennung" ausgewählt. Das hat funktioniert.
Vor ein paar Tagen (evcc Version unklar) habe ich (um etwas zu testen) mein Fahrzeug `db:2` als Standard ausgewählt. Das hat auch funktioniert.
Problem ist, ich komme jetzt nicht mehr zurück auf "automatische Erkennung".
Ich kann es ändern und speichern, aber nach einem Neustart ist weiterhin das Fahrzeug ausgewählt.
Auch eine Änderung auf ein anderes Fahrzeug ist nicht möglich.
 

### Steps to reproduce

.


### Configuration details

```yaml
volker@raspberrypi:~ $ evcc config
[main  ] INFO 2025/04/26 06:12:12 evcc 0.203.3
[main  ] INFO 2025/04/26 06:12:12 using config file: /etc/evcc.yaml
[db    ] INFO 2025/04/26 06:12:12 using sqlite database: /var/lib/evcc/evcc.db
[db    ] TRACE 2025/04/26 06:12:12 SELECT count(*) FROM sqlite_master WHERE type='table' AND name="settings" -1 <nil>
[db    ] TRACE 2025/04/26 06:12:12 SELECT sql FROM sqlite_master WHERE type IN ("table","index") AND tbl_name = "settings" AND sql IS NOT NULL order by type = "table" desc 1 <nil>
[db    ] TRACE 2025/04/26 06:12:12 SELECT * FROM `settings` LIMIT 1 -1 <nil>
[db    ] TRACE 2025/04/26 06:12:12 SELECT * FROM `settings` 154 <nil>
[mqtt  ] INFO 2025/04/26 06:12:12 connecting evcc-232636033 at tcp://localhost:1883
[mqtt  ] DEBUG 2025/04/26 06:12:12 tcp://localhost:1883 connected
[db    ] TRACE 2025/04/26 06:12:12 SELECT count(*) FROM sqlite_master WHERE type='table' AND name="configs" -1 <nil>
[db    ] TRACE 2025/04/26 06:12:12 SELECT sql FROM sqlite_master WHERE type IN ("table","index") AND tbl_name = "configs" AND sql IS NOT NULL order by type = "table" desc 1 <nil>
[db    ] TRACE 2025/04/26 06:12:12 SELECT * FROM `configs` LIMIT 1 -1 <nil>
[db    ] TRACE 2025/04/26 06:12:12 SELECT * FROM `configs` WHERE `configs`.`class` = 1 4 <nil>
charger
---
db:4 {Type:template Title: Icon: Product:} map[host:localhost port:5200 template:daheimladen-mb]
db:6 {Type:template Title: Icon: Product:} map[ain:***** icon:bike integrateddevice:true password:***** standbypower:10 template:fritzdect uri:http://192.168.178.1 user:*****]
db:8 {Type:template Title: Icon: Product:} map[channel:0 heating:true host:192.168.178.29 icon:heatpump integrateddevice:true standbypower:-1 template:shelly]
db:10 {Type:template Title: Icon: Product:} map[channel:0 heating:true host:192.168.178.21 icon:heatpump integrateddevice:true standbypower:10 template:shelly]

[db    ] TRACE 2025/04/26 06:12:12 SELECT * FROM `configs` WHERE `configs`.`class` = 2 2 <nil>
meter
---
db:16 {Type:template Title: Icon: Product:} map[host:192.168.178.22 template:shelly-pro-3em usage:grid]
db:18 {Type:template Title:SolarLog Icon: Product:} map[host:192.168.178.23 port:502 template:solarlog usage:pv]

[db    ] TRACE 2025/04/26 06:12:12 SELECT * FROM `configs` WHERE `configs`.`class` = 3 2 <nil>
vehicle
---
db:2 {Type:template Title: Icon: Product:} map[cache:15m capacity:73 icon:car language:de mode:pv password:***** template:hyundai title:(T)Raumschiff user:*****]
db:17 {Type:template Title: Icon: Product:} map[capacity:73 icon:bus template:offline title:ohne_Bluelink]

[db    ] TRACE 2025/04/26 06:12:12 SELECT * FROM `configs` WHERE `configs`.`class` = 4 0 <nil>
[db    ] TRACE 2025/04/26 06:12:12 SELECT * FROM `configs` WHERE `configs`.`class` = 5 4 <nil>
loadpoint
---
db:5 {Type: Title: Icon: Product:} map[charger:db:4 circuit:load defaultMode:off limitEnergy:0 limitSoc:0 maxCurrent:20 meter: minCurrent:6 mode:pv phasesConfigured:1 planEnergy:0 planTime:0001-01-01T00:00:00Z soc:map[estimate:true poll:map[interval:2.16e+13 mode:charging]] thresholds:map[disable:map[delay:1.2e+12 threshold:0] enable:map[delay:6e+10 threshold:0]] title:Carport(L3) vehicle:db:2]
db:7 {Type: Title: Icon: Product:} map[charger:db:6 circuit:main defaultMode: maxCurrent:1 meter: minCurrent:1 mode:off phasesConfigured:1 priority:1 soc:map[estimate:true poll:map[interval:3.6e+12 mode:charging]] thresholds:map[disable:map[delay:1.8e+11 threshold:0] enable:map[delay:6e+10 threshold:0]] title:Garage vehicle:Vbike]
db:9 {Type: Title: Icon: Product:} map[charger:db:8 circuit:WPSG defaultMode: limitSoc:0 maxCurrent:2 meter: minCurrent:2 mode:off phasesConfigured:1 priority:3 soc:map[estimate:false poll:map[interval:3.6e+12 mode:charging]] thresholds:map[disable:map[delay:3e+11 threshold:0] enable:map[delay:1.2e+12 threshold:0]] title:WP-SG+ vehicle:WasserSG]
db:11 {Type: Title: Icon: Product:} map[charger:db:10 circuit: defaultMode:now limitSoc:99 maxCurrent:2 meter: minCurrent:2 mode:now phasesConfigured:1 priority:5 soc:map[estimate:false poll:map[interval:3.6e+12 mode:charging]] thresholds:map[disable:map[delay:1.8e+11 threshold:0] enable:map[delay:6e+10 threshold:0]] title:WP(L2)-Info- vehicle:WasserWP]

[db    ] TRACE 2025/04/26 06:12:12 SELECT * FROM `configs` WHERE `configs`.`class` = 6 0 <nil>
```

### Log details

```text
keine Ahnung, welches Log ich hier einbringen soll. Bitte um entsprechende Hinweise.
```

### What type of operating system or environment does evcc run on?

Linux

### External automation

- [x] I have made sure that no external automation like HomeAssistant or Node-RED is active or accessing any of the mentioned devices when this issue occurs.

### Nightly build

- [x] I have verified that the issue is reproducible with the latest nightly build

### Version

0.203.3

GORM oder Compiler bug?

> ../../../go/pkg/mod/gorm.io/gorm@v1.26.0/internal/stmt_store/stmt_store.go:99:10: cannot use defaultMaxSize (untyped int constant 9223372036854775807) as int value in assignment (overflows)

--- a/assets/js/components/Config/CircuitsModal.vue
+++ b/assets/js/components/Config/CircuitsModal.vue
@@ -7,6 +7,7 @@
 		:defaultYaml="defaultYaml"
 		removeKey="circuits"
 		endpoint="/config/circuits"
+		data-testid="circuits-modal"
 		@changed="$emit('changed')"
 	/>
 </template>

--- a/cmd/setup.go
+++ b/cmd/setup.go
@@ -1007,7 +1007,7 @@ CONTINUE:
 		}
 
 		if !isRoot && !instance.HasMeter() {
-			return fmt.Errorf("circuit %s has no meter and no loadpoint assigned", dev.Config().Name)
+			log.INFO.Printf("circuit %s has no meter and no loadpoint assigned", dev.Config().Name)
 		}
 	}
 

--- a/cmd/setup_circuits_test.go
+++ b/cmd/setup_circuits_test.go
@@ -102,8 +102,7 @@ loadpoints:
 
 	// circuit without device
 	err = validateCircuits(lps)
-	suite.Require().Error(err)
-	suite.Require().Equal("circuit slave has no meter and no loadpoint assigned", err.Error())
+	suite.Require().NoError(err)
 }
 
 func (suite *circuitsTestSuite) TestMissingRootCircuit() {

--- a/i18n/de.json
+++ b/i18n/de.json
@@ -193,6 +193,7 @@
       "chargerPowerCustomHelp": "Definiere einen eigenen Strombereich.",
       "chargerTypeLabel": "Ladetyp",
       "chargingTitle": "Laden",
+      "circuitHelp": "Lastmanagement-Zuordnung, um die Leistungs- und Stromgrenzen nicht zu überschreiten.",
       "circuitLabel": "Stromkreis",
       "circuitUnassigned": "nicht zugewiesen",
       "defaultModeHelp": "Lademodus beim Anschließen des Fahrzeugs.",

--- a/i18n/en.json
+++ b/i18n/en.json
@@ -193,6 +193,7 @@
       "chargerPowerCustomHelp": "Define a custom current range.",
       "chargerTypeLabel": "Charger type",
       "chargingTitle": "Charging",
+      "circuitHelp": "Load management assignment to ensure power and current limits are not exceeded.",
       "circuitLabel": "Circuit",
       "circuitUnassigned": "unassigned",
       "defaultModeHelp": "Charging mode when connecting the vehicle.",

--- a/server/http_config_loadpoint_handler.go
+++ b/server/http_config_loadpoint_handler.go
@@ -206,7 +206,7 @@ func updateLoadpointHandler() http.HandlerFunc {
 
 		// merge here to maintain dynamic part of the config
 		other := configurable.Config().Other
-		if err := mergo.Merge(&other, static); err != nil {
+		if err := mergo.Merge(&other, static, mergo.WithOverride); err != nil {
 			jsonError(w, http.StatusBadRequest, err)
 			return
 		}

--- a/tests/config-circuit.evcc.yaml
+++ b/tests/config-circuit.evcc.yaml
@@ -1,0 +1,30 @@
+site:
+  meters:
+    grid: grid
+
+meters:
+  - name: grid
+    type: template
+    template: demo-meter
+    power: 2070
+    currentL1: 3
+    currentL2: 3
+    currentL3: 3
+
+loadpoints:
+  - title: Carport
+    charger: charger
+    circuit: main
+
+circuits:
+  - name: main
+    meter: grid
+    maxcurrent: 16
+
+chargers:
+  - name: charger
+    type: template
+    template: demo-charger
+    status: C
+    enabled: true
+    power: 1000

--- a/tests/config-circuit.spec.js
+++ b/tests/config-circuit.spec.js
@@ -2,7 +2,7 @@ import { test, expect } from "@playwright/test";
 import { start, stop, baseUrl } from "./evcc";
 import { enableExperimental } from "./utils";
 
-const CONFIG_YAML = "config-circuit.yaml";
+const CONFIG_YAML = "config-circuit.evcc.yaml";
 
 test.use({ baseURL: baseUrl() });
 

--- a/tests/config-circuit.spec.js
+++ b/tests/config-circuit.spec.js
@@ -1,8 +1,15 @@
 import { test, expect } from "@playwright/test";
-import { start, stop, baseUrl } from "./evcc";
-import { enableExperimental } from "./utils";
+import { start, stop, restart, baseUrl } from "./evcc";
+import {
+  enableExperimental,
+  expectModalVisible,
+  expectModalHidden,
+  editorClear,
+  editorType,
+} from "./utils";
 
 const CONFIG_YAML = "config-circuit.evcc.yaml";
+const CONFIG_EMPTY = "config-empty.evcc.yaml";
 
 test.use({ baseURL: baseUrl() });
 
@@ -30,6 +37,139 @@ test.describe("circuit", async () => {
     await expect(page.getByTestId("circuits")).toContainText(["Power", "2.1 kW"].join(""));
     await expect(page.getByTestId("circuits")).toContainText(
       ["Current", "3.0 A / 16.0 A"].join("")
+    );
+  });
+
+  test("via ui", async ({ page }) => {
+    await start(CONFIG_EMPTY);
+
+    await page.goto("/#/config");
+    await enableExperimental(page);
+
+    // add grid meter
+    await page.getByRole("button", { name: "Add grid meter" }).click();
+    const meterModal = page.getByTestId("meter-modal");
+    await expectModalVisible(meterModal);
+    await meterModal.getByLabel("Manufacturer").selectOption("Demo meter");
+    await meterModal.getByLabel("Power").fill("2070");
+    await page.getByRole("button", { name: "Show advanced settings" }).click();
+    await meterModal.getByLabel("L1 current").fill("3");
+    await meterModal.getByLabel("L2 current").fill("3");
+    await meterModal.getByLabel("L3 current").fill("3");
+    await meterModal.getByRole("button", { name: "Save" }).click();
+    await expectModalHidden(meterModal);
+
+    // add loadpoint and charger
+    const lpModal = page.getByTestId("loadpoint-modal");
+    await page.getByRole("button", { name: "Add charge point" }).click();
+    await expectModalVisible(lpModal);
+    await lpModal.getByLabel("Title").fill("Carport");
+
+    // add charger
+    await lpModal.getByRole("button", { name: "Add charger" }).click();
+    const chargerModal = page.getByTestId("charger-modal");
+    await expectModalVisible(chargerModal);
+    await chargerModal.getByLabel("Manufacturer").selectOption("Demo charger");
+    await chargerModal.getByLabel("Charge status").selectOption("C");
+    await chargerModal.getByLabel("Power").fill("1000");
+    await chargerModal.getByRole("button", { name: "Save" }).click();
+    await expectModalHidden(chargerModal);
+    await expectModalVisible(lpModal);
+
+    await lpModal.getByRole("button", { name: "Save" }).click();
+    await expectModalHidden(lpModal);
+
+    // add circuit via ui as yaml input
+    await page.getByTestId("circuits").getByRole("button", { name: "edit" }).click();
+    const circuitsModal = page.getByTestId("circuits-modal");
+    await expectModalVisible(circuitsModal);
+
+    const editor = circuitsModal.getByTestId("yaml-editor");
+    await editorClear(editor);
+    await editorType(editor, [
+      // prettier-ignore
+      "- name: main",
+      "  meter: db:1",
+      "maxcurrent: 16",
+      "Shift+Tab",
+      "- name: house",
+      "  title: House",
+      "maxcurrent: 10",
+      "parent: main",
+      "Shift+Tab",
+      "- name: garage",
+      "  title: Garage",
+      "maxcurrent: 8",
+      "parent: main",
+    ]);
+
+    await circuitsModal.getByRole("button", { name: "Save" }).click();
+    await expectModalHidden(circuitsModal);
+
+    // restart
+    const restartButton = await page
+      .getByTestId("bottom-banner")
+      .getByRole("button", { name: "Restart" });
+    await expect(restartButton).toBeVisible();
+    await restart(CONFIG_EMPTY);
+    await page.reload();
+
+    // assign loadpoint to circuit
+    await page.getByTestId("loadpoint").getByRole("button", { name: "edit" }).click();
+    await expectModalVisible(lpModal);
+    await lpModal.getByLabel("Circuit").selectOption("Garage [garage]");
+    await lpModal.getByLabel("Circuit").selectOption("House [house]");
+    await lpModal.getByRole("button", { name: "Save" }).click();
+    await expectModalHidden(lpModal);
+
+    // save, restart and check values
+    await expect(restartButton).toBeVisible();
+    await restart(CONFIG_EMPTY);
+    await page.reload();
+
+    // verify the configuration matches the yaml test
+    await expect(page.getByTestId("loadpoint")).toHaveCount(1);
+    await expect(page.getByTestId("loadpoint")).toContainText(["Power", "1.0 kW"].join(""));
+
+    await expect(page.getByTestId("grid")).toHaveCount(1);
+    await expect(page.getByTestId("grid")).toContainText(["Power", "2.1 kW"].join(""));
+    await expect(page.getByTestId("grid")).toContainText(
+      ["Current L1, L2, L3", "3.0 · 3.0 · 3.0 A"].join("")
+    );
+
+    await expect(page.getByTestId("circuits")).toHaveCount(1);
+    await expect(page.getByTestId("circuits")).toContainText(
+      ["(main)", "Power", "2.1 kW", "Current", "3.0 A / 16.0 A"].join("")
+    );
+    await expect(page.getByTestId("circuits")).toContainText(
+      ["House (house)", "Power", "1.0 kW", "Current", "6.0 A / 10.0 A"].join("")
+    );
+    await expect(page.getByTestId("circuits")).toContainText(
+      ["Garage (garage)", "Power", "0.0 kW", "Current", "0.0 A / 8.0 A"].join("")
+    );
+
+    // assign to garage
+    await page.getByTestId("loadpoint").getByRole("button", { name: "edit" }).click();
+    await expectModalVisible(lpModal);
+    await lpModal.getByLabel("Circuit").selectOption("Garage [garage]");
+    await lpModal.getByRole("button", { name: "Save" }).click();
+    await expectModalHidden(lpModal);
+
+    // save, restart and check values
+    await expect(restartButton).toBeVisible();
+    await restart(CONFIG_EMPTY);
+    await page.reload();
+
+    // verify circuits
+    await expect(page.getByTestId("circuits")).toHaveCount(1);
+    await expect(page.getByTestId("circuits")).toContainText(
+      ["(main)", "Power", "2.1 kW", "Current", "3.0 A / 16.0 A"].join("")
+    );
+    await expect(page.getByTestId("circuits")).toContainText(
+      ["House (house)", "Power", "0.0 kW", "Current", "0.0 A / 10.0 A"].join("")
+    );
+    await expect(page.getByTestId("circuits")).toContainText(
+      ["Garage (garage)", "Power", "1.0 kW", "Current", "6.0 A / 8.0 A"].join("")
     );
   });
 });

--- a/tests/config-circuit.spec.js
+++ b/tests/config-circuit.spec.js
@@ -1,0 +1,35 @@
+import { test, expect } from "@playwright/test";
+import { start, stop, baseUrl } from "./evcc";
+import { enableExperimental } from "./utils";
+
+const CONFIG_YAML = "config-circuit.yaml";
+
+test.use({ baseURL: baseUrl() });
+
+test.afterEach(async () => {
+  await stop();
+});
+
+test.describe("circuit", async () => {
+  test("from yaml", async ({ page }) => {
+    await start(CONFIG_YAML);
+
+    await page.goto("/#/config");
+    await enableExperimental(page);
+
+    await expect(page.getByTestId("loadpoint")).toHaveCount(1);
+    await expect(page.getByTestId("loadpoint")).toContainText(["Power", "1.0 kW"].join(""));
+
+    await expect(page.getByTestId("grid")).toHaveCount(1);
+    await expect(page.getByTestId("grid")).toContainText(["Power", "2.1 kW"].join(""));
+    await expect(page.getByTestId("grid")).toContainText(
+      ["Current L1, L2, L3", "3.0 · 3.0 · 3.0 A"].join("")
+    );
+
+    await expect(page.getByTestId("circuits")).toHaveCount(1);
+    await expect(page.getByTestId("circuits")).toContainText(["Power", "2.1 kW"].join(""));
+    await expect(page.getByTestId("circuits")).toContainText(
+      ["Current", "3.0 A / 16.0 A"].join("")
+    );
+  });
+});

